Branch-SDK/src/io/branch/indexing/BranchUniversalObject.java

```diff
@@ -80,6 +105,8 @@ public BranchUniversalObject() {
         type_ = "";
         indexMode_ = CONTENT_INDEX_MODE.PUBLIC; // Default content indexing mode is public
         expirationInMilliSec_ = 0L;
+        price_ = 0.0;
+        currency_ = CURRENCY_TYPE.NONE;
```

Comment:
honestly.. I'm not sure why any of those values `canonicalIdentifier`, `canonicalUrl`, etc should be initialized to an empty string?